MovieApp/AppViewController.swift

```diff
@@ -17,49 +34,112 @@ class AppViewController: UIViewController {
+        view.addSubview(errorView)
+        errorView.snp.makeConstraints{ (make) in
+            make.top.leading.trailing.bottom.equalToSuperview()
+        }
+        
+        
+        let errorTextView = UITextView()
+        errorTextView.backgroundColor = UIColor(red: 0.91, green: 0.93, blue: 0.96, alpha: 1.00)
+        errorTextView.text = "Network error: something went wrong :("
+        errorTextView.font = UIFont.systemFont(ofSize: 18)
+        errorView.addSubview(errorTextView)
+        
+        errorTextView.snp.makeConstraints{ (make) in
+            make.centerY.centerX.equalToSuperview()
+            make.width.equalTo(300)
+            make.height.equalTo(200)
+        }
+    }
+    
+    
+    private func buildNavBar() {
+        let navigationBarAppearance = UINavigationBarAppearance()
+        navigationBarAppearance.configureWithDefaultBackground()
+        navigationBarAppearance.backgroundColor = UIColor(red: 0.04, green: 0.15, blue: 0.25, alpha: 1.00)
+
+        navigationItem.scrollEdgeAppearance = navigationBarAppearance
+        navigationItem.standardAppearance = navigationBarAppearance
+        navigationItem.compactAppearance = navigationBarAppearance
+    }
+    
+    
     private func buildViews() {
-        scrollView.addSubview(mainView)
         view.addSubview(scrollView)
         view.backgroundColor = .white
-        scrollView.bounces = false
+
         
-        scrollView.translatesAutoresizingMaskIntoConstraints = false
-        scrollView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
-        scrollView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor).isActive = true
-        scrollView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor).isActive = true
-        scrollView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+        scrollView.snp.makeConstraints { (make) in
+            make.top.bottom.equalToSuperview()
+            make.leading.equalTo(view.safeAreaLayoutGuide.snp.leading)
+            make.trailing.equalTo(view.safeAreaLayoutGuide.snp.trailing)
+        }
+    
         
+        createImageView()
         
+        createCollectionView()
         
-        mainView.translatesAutoresizingMaskIntoConstraints = false
-        mainView.topAnchor.constraint(equalTo: scrollView.topAnchor).isActive = true
-        mainView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor).isActive = true
-        mainView.widthAnchor.constraint(equalTo: scrollView.widthAnchor).isActive = true
-
+        let navBar = UINavigationBarAppearance()
+        navBar.backgroundColor = UIColor(red: 0.04, green: 0.15, blue: 0.25, alpha: 1.00)
         
-        createImageView()
+        self.navigationController?.navigationBar.scrollEdgeAppearance = navBar
         
-        createCollectionView()
+        
+        let titleView = UIView()
```

Comment:
You are building this navigation bar view in multiple places. You should have this built only once. Maybe make a NavigationTitleView and put this code in that view and then just init that view where needed.